### PR TITLE
Minor (non breaking) docs and naming changes

### PR DESCRIPTION
**Pull request by MathisGD**

### CODE REVIEW

File: src/interfaces/IDelegates.sol [hunk 1]
```diff
@@ -9,20 +9,20 @@ interface IDelegates {
...
     event DelegateVotesChanged(address indexed delegate, uint256 previousVotes, uint256 newVotes);
 
-    // @dev Returns the current amount of votes that `account` has.
-    function delegatedVotingPower(address account) external view returns (uint256);
+    // @dev Returns the current amount of votes that `delegator` has.
+    function delegatedVotingPower(address delegator) external view returns (uint256);
 
-    // @dev Returns the delegate that `account` has chosen.
-    function delegates(address account) external view returns (address);
+    // @dev Returns the delegate that `delegator` has chosen.
```
Comment (QGarchery) [inline src/interfaces/IDelegates.sol:24]:
```suggestion
    // @dev Returns the delegatee that `delegator` has chosen.
```

File: src/ERC20DelegatesUpgradeable.sol [hunk 2]
```diff
@@ -86,27 +92,25 @@ abstract contract ERC20DelegatesUpgradeable is Initializable, ERC20Upgradeable,
...
+        emit DelegateChanged(delegator, oldDelegate, delegatee);
+        _moveDelegateVotes(oldDelegate, delegatee, _getVotingUnits(delegator));
     }
 
-    /// @dev Must return the voting units held by an account.
-    function _getVotingUnits(address account) internal view returns (uint256) {
-        return balanceOf(account);
+    /// @dev Must return the voting units held by an delegator.
+    function _getVotingUnits(address delegator) internal view returns (uint256) {
+        return balanceOf(delegator);
```
Comment (QGarchery) [inline src/ERC20DelegatesUpgradeable.sol:107]:
```suggestion
    /// @dev Returns the voting units held by a delegator.
    function _getVotingUnits(address delegator) internal view returns (uint256) {
        return balanceOf(delegator);
```

**Review by QGarchery — APPROVED** [review]

File: src/interfaces/IERC20DelegatesUpgradeable.sol [hunk 3]
```diff
@@ -22,7 +22,7 @@ interface IDelegates {
     function delegatedVotingPower(address delegator) external view returns (uint256);
 
     // @dev Returns the delegate that `delegator` has chosen.
-    function delegatee(address delegator) external view returns (address);
+    function delegates(address delegator) external view returns (address);
```
Comment (QGarchery) [inline src/interfaces/IERC20DelegatesUpgradeable.sol:25]: This is a breaking change though

File: src/ERC20DelegatesUpgradeable.sol [hunk 4]
```diff
@@ -86,27 +92,25 @@ abstract contract ERC20DelegatesUpgradeable is Initializable, ERC20Upgradeable,
...
+    /// @dev Returns the voting units held by a delegator.
+    function _getVotingUnits(address delegator) internal view returns (uint256) {
+        return balanceOf(delegator);
     }
 
     /// @dev Moves voting power when tokens are transferred.
     /// @dev Emits a {IDelegates-DelegateVotesChanged} event.
     function _update(address from, address to, uint256 value) internal virtual override {
         super._update(from, to, value);
-        // No check of supply cap here like in OZ implementation as MORPHO has a 1B total supply cap.
```
Comment (adhusson) [inline src/ERC20DelegatesUpgradeable.sol:109]: A comment somewhere on the topic would be good, maybe the README?

**Review by adhusson — APPROVED** [review]

File: src/ERC20DelegatesUpgradeable.sol (hunk 4 above)
Comment (MerlinEgalite) [reply src/ERC20DelegatesUpgradeable.sol:109]: I think we should keep this comment

File: src/ERC20DelegatesUpgradeable.sol [hunk 5]
```diff
@@ -36,39 +41,40 @@ abstract contract ERC20DelegatesUpgradeable is Initializable, ERC20Upgradeable,
...
         ERC20DelegatesStorage storage $ = _getERC20DelegatesStorage();
-        return $._delegatee[account];
+        return $._delegatee[delegator];
     }
 
-    /* EXTERNAL */
-
-    /// @dev Returns the current amount of votes delegated to `account`.
-    function delegatedVotingPower(address account) external view returns (uint256) {
+    /// @dev Returns the current amount of votes delegated to `delegator`.
```
Comment (colin-morpho) [inline src/ERC20DelegatesUpgradeable.sol:57]:
```suggestion
    /// @dev Returns the current voting power delegated to `delegator`.
```

Comment (colin-morpho) [inline src/ERC20DelegatesUpgradeable.sol:57]:
```suggestion
    /// @dev Returns the current amount of votes delegated by the `delegator`.
```

File: src/ERC20DelegatesUpgradeable.sol [hunk 6]
```diff
@@ -36,39 +41,40 @@ abstract contract ERC20DelegatesUpgradeable is Initializable, ERC20Upgradeable,
...
+    /// @dev Delegates the balance of the sender to `delegatee`.
     function delegate(address delegatee) external {
-        address account = _msgSender();
-        _delegate(account, delegatee);
+        address delegator = _msgSender();
+        _delegate(delegator, delegatee);
     }
 
-    /// @dev Delegates votes from signer to `delegatee`.
+    /// @dev Delegates the balance of the signer to `delegatee`.
```
Comment (colin-morpho) [inline src/ERC20DelegatesUpgradeable.sol:77]:
```suggestion
    /// @dev Delegates the balance of the signer to the `delegatee`.
```

File: src/ERC20DelegatesUpgradeable.sol [hunk 7]
```diff
@@ -36,39 +41,40 @@ abstract contract ERC20DelegatesUpgradeable is Initializable, ERC20Upgradeable,
...
     }
 
-    /// @dev Delegates votes from the sender to `delegatee`.
+    /* DELEGATE */
+
+    /// @dev Delegates the balance of the sender to `delegatee`.
     function delegate(address delegatee) external {
-        address account = _msgSender();
-        _delegate(account, delegatee);
+        address delegator = _msgSender();
```
Comment (MerlinEgalite) [inline src/ERC20DelegatesUpgradeable.sol:73]: Imo it makes comparison more difficult with OZ code, so not sure it's helpful

**Review by colin-morpho — APPROVED** [review]

File: src/ERC20DelegatesUpgradeable.sol [hunk 8]
```diff
@@ -36,39 +41,40 @@ abstract contract ERC20DelegatesUpgradeable is Initializable, ERC20Upgradeable,
...
-        return $._delegatee[account];
+        return $._delegatee[delegator];
     }
 
-    /* EXTERNAL */
-
-    /// @dev Returns the current amount of votes delegated to `account`.
-    function delegatedVotingPower(address account) external view returns (uint256) {
+    /// @dev Returns the current amount of votes delegated to `delegator`.
+    function delegatedVotingPower(address delegator) external view returns (uint256) {
```
Comment (Jean-Grimal) [inline src/ERC20DelegatesUpgradeable.sol:58]:
```suggestion
    /// @dev Returns the current amount of votes delegated to `delegatee`.
    function delegatedVotingPower(address delegatee) external view returns (uint256) {
```

File: src/ERC20DelegatesUpgradeable.sol [hunk 9]
```diff
@@ -36,39 +41,40 @@ abstract contract ERC20DelegatesUpgradeable is Initializable, ERC20Upgradeable,
...
         ERC20DelegatesStorage storage $ = _getERC20DelegatesStorage();
-        return $._delegatedVotingPower[account];
+        return $._delegatedVotingPower[delegator];
```
Comment (Jean-Grimal) [inline src/ERC20DelegatesUpgradeable.sol:60]:
```suggestion
        return $._delegatedVotingPower[delegatee];
```

File: src/ERC20DelegatesUpgradeable.sol (hunk 7 above)
Comment (Jean-Grimal) [reply src/ERC20DelegatesUpgradeable.sol:73]: I agree

**Review by Jean-Grimal — COMMENTED** [review]
I would rather keep "account"

File: src/ERC20DelegatesUpgradeable.sol (hunk 4 above)
Comment (MathisGD) [reply src/ERC20DelegatesUpgradeable.sol:109]: OZ checks the totalSupply on mint, so why should we speak about this?

```solidity
        if (from == address(0)) {
            // Overflow check required: The rest of the code assumes that totalSupply never overflows
            _totalSupply += value;
```

File: src/ERC20DelegatesUpgradeable.sol (hunk 5 above)
Comment (MathisGD) [reply src/ERC20DelegatesUpgradeable.sol:57]: it's to `delegatee`, my mistake

File: src/ERC20DelegatesUpgradeable.sol [hunk 10]
```diff
@@ -78,16 +78,16 @@ abstract contract ERC20DelegatesUpgradeable is
...
             _hashTypedDataV4(keccak256(abi.encode(DELEGATION_TYPEHASH, delegatee, nonce, expiry))), v, r, s
         );
 
         ERC20DelegatesStorage storage $ = _getERC20DelegatesStorage();
-        uint256 current = $._delegationNonce[signer];
-        require(nonce == current, InvalidDelegationNonce(signer, current));
-        $._delegationNonce[signer]++;
+        uint256 current = $._delegationNonce[delegator];
+        require(nonce == current, InvalidDelegationNonce(delegator, current));
+        $._delegationNonce[delegator]++;
```
Comment (QGarchery) [inline src/ERC20DelegatesUpgradeable.sol:88]:
```suggestion
        require(nonce == $._delegationNonce[delegator]++, InvalidDelegationNonce(delegator, current));
```
optional

**Review by QGarchery — DISMISSED** [review]

File: src/ERC20DelegatesUpgradeable.sol (hunk 10 above)
Comment (MathisGD) [reply src/ERC20DelegatesUpgradeable.sol:88]: doesn't work because of the error, but tbh i think that logging the nonces is not super useful

Comment (MathisGD) [reply src/ERC20DelegatesUpgradeable.sol]: I removed the nonce log in the error

File: src/ERC20DelegatesUpgradeable.sol [hunk 11]
```diff
@@ -36,39 +41,40 @@ abstract contract ERC20DelegatesUpgradeable is Initializable, ERC20Upgradeable,
 
     /// @custom:storage-location erc7201:morpho.storage.ERC20Delegates
     struct ERC20DelegatesStorage {
-        mapping(address account => address) _delegatee;
+        mapping(address delegator => address) _delegatee;
         mapping(address delegatee => uint256) _delegatedVotingPower;
-        mapping(address account => uint256) _delegationNonce;
+        mapping(address delegator => uint256) _delegationNonce;
```
[the remark this answers is not in the document]
Comment (MathisGD) [reply src/ERC20DelegatesUpgradeable.sol:46]: tbh I'm hesitant to remove the names, sometimes it's better

**Review by colin-morpho — APPROVED** [review]

File: src/ERC20DelegatesUpgradeable.sol (hunk 7 above)
Comment (QGarchery) [reply src/ERC20DelegatesUpgradeable.sol:73]: To me we don't have to comply to OZ coding style, and if we find a better (more appropriate for us) way, then we should do it. Here the naming is more specified and can still be consistent, by using delegator/delegatee everywhere, so I find it better

**Review by Jean-Grimal — APPROVED** [review]

File: src/interfaces/IERC20DelegatesUpgradeable.sol [hunk 12]
```diff
@@ -5,24 +5,24 @@ pragma solidity >=0.5.0;
...
 
-    // @dev Returns the current amount of votes that `account` has.
-    function delegatedVotingPower(address account) external view returns (uint256);
+    // @dev Returns the current amount of votes that `delegator` has.
+    function delegatedVotingPower(address delegator) external view returns (uint256);
 
-    // @dev Returns the delegate that `account` has chosen.
-    function delegates(address account) external view returns (address);
+    // @dev Returns the delegatee that `delegator` has chosen.
+    function delegates(address delegator) external view returns (address);
```
Comment (MerlinEgalite) [inline src/interfaces/IERC20DelegatesUpgradeable.sol:25]: not updated

**Review by MerlinEgalite — CHANGES_REQUESTED** [review]

**Review by Jean-Grimal — APPROVED** [review]

File: src/interfaces/IERC20DelegatesUpgradeable.sol [hunk 13]
```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity >=0.5.0;
+
+/// @title IDelegates
+/// @author Morpho Labs
```
Comment (MerlinEgalite) [inline src/interfaces/IERC20DelegatesUpgradeable.sol:5]:
```suggestion
/// @author Morpho Association
```

**Review by Jean-Grimal — APPROVED** [review]

**Review by colin-morpho — APPROVED** [review]

**Review by QGarchery — APPROVED** [review]

**Pull request merged**